src/ProductView/TabBar/QueryInterface/QueryInterface.js

```diff
@@ -0,0 +1,143 @@
+        this.handleChangeQuery = this.handleChangeQuery.bind(this);
+        this.handleSaveQuery = this.handleSaveQuery.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({ eventQuery: this.props.defaultQuery });
+    }
+
+    addEmptyAttribute(attributes) {
+        attributes = attributes.concat([{
+            'id': this.nextAttributeId,
+            "name": "",
+            "type": "",
+            "readonly": false
+        }]);
+        this.nextAttributeId = this.nextAttributeId + 1;
+        return attributes;
+    }
+
+    getEventType() {
+        const eventTypeAttributes = {};
+        const attributes = this.state.eventTypeAttributes;
+        for (let i = 0; i < attributes.length; i++) {
+            if (attributes[i].name) {
+                eventTypeAttributes[attributes[i].name] = attributes[i].type;
+            }
+        }
+        const eventType = {
+            'name': this.state.eventTypeName,
+            'timestamp': 'timestamp',
+            'attributes': eventTypeAttributes
+        };
+        return eventType;
+    }
+
+    getEventTypeAttribute(id) {
+        return this.state.eventTypeAttributes.find((attribute) => {
+            return attribute.id === id;
+        });
+    }
+
+    handleChangeEventTypeName(name) {
+        this.setState({ eventTypeName: name });
+    }
+
+    handleChangeAttributeName(id, name) {
+        const currentAttribute = this.getEventTypeAttribute(id);
+        if (currentAttribute.readonly) {
+            return;
+        }
+
+        let attributes = this.state.eventTypeAttributes;
+        const currentIndex = attributes.indexOf(currentAttribute);
+        attributes[currentIndex].name = name;
+
+        if (!name) {
+            attributes.splice(currentIndex, 1);
+        }
+        else if (currentIndex === attributes.length - 1) {
+            attributes = this.addEmptyAttribute(attributes);
+        }
+        this.setState({ eventTypeAttributes: attributes });
+    }
+
+    handleChangeAttributeType(id, type) {
+        const currentAttribute = this.getEventTypeAttribute(id);
+        if (currentAttribute.readonly) {
+            return;
+        }
+
+        const attributes = this.state.eventTypeAttributes;
+        const currentIndex = attributes.indexOf(currentAttribute);
+        attributes[currentIndex].type = type;
+        this.setState({ eventTypeAttributes: attributes });
+    }
+
+    handleChangeQuery(event) {
+        this.setState({ eventQuery: event.target.value });
+    }
+
+    handleSaveQuery() {
+        const eventType = this.getEventType();
+        this.props.dataSender.createEventtype(this.state.eventQuery, eventType, this.handleEventTypeData, this.handleError);
+    }
+
+    render() {
+        return (
+            <div className="query-interface">
+                <Modal onSave={this.handleSaveQuery}>
+                    <QueryInterfaceHeader eventTypeName={this.state.eventTypeName}
```

Comment:
Maybe rename header to attributeList or something elese. Header indicates an protocol header or an UI header ...